plugins/default/src/Ide/Plugin/ModuleName.hs

```diff
@@ -0,0 +1,162 @@
+                                             defaultPluginDescriptor)
+import           Language.Haskell.LSP.Core  (LspFuncs, getVirtualFileFunc)
+import           Language.Haskell.LSP.Types (ApplyWorkspaceEditParams (..),
+                                             CAResult (CACodeAction),
+                                             CodeAction (CodeAction),
+                                             CodeActionKind (CodeActionQuickFix),
+                                             CodeLens (CodeLens),
+                                             CodeLensParams (CodeLensParams),
+                                             Command (Command),
+                                             ServerMethod (..),
+                                             TextDocumentIdentifier (TextDocumentIdentifier),
+                                             TextEdit (TextEdit), Uri,
+                                             WorkspaceEdit (..),
+                                             uriToNormalizedFilePath)
+import           Language.Haskell.LSP.VFS   (virtualFileText)
+import           System.FilePath            (dropExtension)
+
+-- |Plugin descriptor
+descriptor :: PluginId -> PluginDescriptor
+descriptor plId =
+  (defaultPluginDescriptor plId)
+    { pluginId = plId,
+        pluginCodeLensProvider = Just codeLens
+       ,pluginCommands = [PluginCommand editCommandName editCommandName editCmd]
+        -- pluginCodeActionProvider = Just codeAction
+    }
+
+-- | Generate code lenses
+codeLens :: LspFuncs c -> IdeState -> PluginId -> CodeLensParams -> IO (Either a2 (List CodeLens))
+codeLens lsp state pluginId (CodeLensParams (TextDocumentIdentifier uri) _) = do
+  pid <- getPid
+  actions (asCodeLens (mkLspCmdId pid pluginId editCommandName)) lsp state uri
+
+-- | Generate code actions.
+-- NOTE: Not invoked on an empty module (but codeLens is, why?)
+codeAction :: LspFuncs c -> IdeState -> p1 -> TextDocumentIdentifier -> p2 -> p3 -> IO (Either a (List CAResult))
+codeAction lsp state _plId (TextDocumentIdentifier uri) _range _ = actions asCodeAction lsp state uri
+
+-- Copied from "Ide.Plugin"
+mkLspCmdId :: T.Text -> PluginId -> CommandId -> T.Text
+mkLspCmdId pid (PluginId plid) (CommandId cid)
+  = pid <> ":" <> plid <> ":" <> cid
+
+editCommandName :: IsString p => p
+editCommandName = "edit"
+
+-- | Generic command to apply a group of edits
+editCmd :: CommandFunction WorkspaceEdit
+editCmd _lf _ide workspaceEdits = return (Right Null, Just $ (WorkspaceApplyEdit,ApplyWorkspaceEditParams workspaceEdits))
+
+-- | Required actions (actually, at most one) that can be converted to either CodeLenses or CodeActions
+actions :: Show a1 => (Action -> a1) -> LspFuncs c -> IdeState -> Uri -> IO (Either a2 (List a1))
+actions convert lsp state uri = do
+    let Just nfp = uriToNormalizedFilePath $ toNormalizedUri uri
+    let Just fp = uriToFilePath' uri
+
+    contents <- liftIO $ getVirtualFileFunc lsp $ toNormalizedUri uri
```

Comment:
Not sure if it is related with the lack of functionality in windows but i had to replace `getVirtualFileFunc lsp` with `getFileContents` in hlint plugin to make it work in windows.
Maybe we should open an issue about :worried:

Reply:
So how should we handle this?

Reply:
https://github.com/tittoassini/haskell-language-server/pull/1 😃 
But i've test that version and the plugin does not work in my windows machine 😟